source/beamline/CLS/CLSShutters.cpp

```diff
@@ -14,7 +19,88 @@ CLSShutters::CLSShutters(const QString &name, QObject *parent) :
 
 CLSShutters::~CLSShutters()
 {
+	shuttersBeamOnOrderMap_.clear();
+}
+
+AMListAction3* CLSShutters::createBeamOnActionList()
+{
+	// create the action list to move the Shutters (sequentially) and wait for the move done
+	AMListAction3 *openShuttersActionList = new AMListAction3(new AMListActionInfo3("Open shutters action list", "Open Shutters"), AMListAction3::Sequential);
+	AMListAction3 *waitShuttersOpenActionList = new AMListAction3(new AMListActionInfo3("Wait shutters open action list", "Wait Shutters Open"), AMListAction3::Parallel);
+
+	// this is to make sure all the controls are checked
+	int currentBeamOnOrder = 1;
+	int checkedControlCount = 0;
+	while (checkedControlCount < shuttersBeamOnOrderMap_.count()) {
```

Comment:
@dretrex I want to make sure they are ordered. And yes, I can use that. Just checked the documents:
```
QList<Key> QMap::keys() const
Returns a list containing all the keys in the map in ascending order.
```